controllers/storagerequest/storagerequest_controller.go

```diff
@@ -441,6 +441,7 @@ func (r *StorageRequestReconciler) reconcileCephFilesystemSubVolumeGroup() error
 
 		r.cephFilesystemSubVolumeGroup.Spec = rookCephv1.CephFilesystemSubVolumeGroupSpec{
 			FilesystemName: cephFilesystem.Name,
+			DataPoolName:   dataPoolValue,
```

Comment:
>we should set dataPoolName when creating cephfssvg, without it we will not be able to use different data pools for different claims stating storageprofile.

Do you see any problem with the existing code?

we can do this but this is an extra where we are allocating the Datapool to the SVG itself. if we do this we can remove the dataPool from the storageclass itself. can you please check it?

Reply:
I did mention the problem in the commit msg, to expand, svg corresponding to a storagerequest which has a custom storageprofile is not being applied w/o this code.

Unfortunately, I don't get the question on storageclasses, do you mean to say even if we create svg w/ a different data pool we can supply another pool name on storageclasses and that'll be used for pvc?

Do you see any discrepancy in https://github.com/red-hat-storage/ocs-operator/pull/2649#issuecomment-2142044038? label vs spec. Is that expected?

Reply:
> How did you verify this one
- No, I observed that spec.datapoolName is empty and saw it has default pool being used in Ceph command and thought it was a mistake.